sdk/runanywhere-python/runanywhere/server/app.py

```
@@ -0,0 +1,643 @@
+
+def _reject_ssrf(url: str) -> None:
+    """Reject an image URL whose host resolves to a private/loopback/link-local/reserved
+    address — a network request must not be usable to reach internal services or cloud
+    metadata (SSRF)."""
+    host = urllib.parse.urlparse(url).hostname
+    if not host:
+        raise SDKException.invalid_input("invalid image URL")
+    try:
+        infos = socket.getaddrinfo(host, None)
+    except OSError as exc:
+        raise SDKException.invalid_input(f"could not resolve image host: {exc}")
```

Ruff (0.15.21)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)